components/style/values/specified/length.rs

```diff
@@ -1226,3 +1227,97 @@ impl LengthOrNumber {
         }
     }
 }
+
+/// A value suitable for a `min-width` or `min-height` property.
+/// Unlike `max-width` or `max-height` properties, a MinLength can be
+/// `auto`, and cannot be `none`.
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+#[allow(missing_docs)]
+pub enum MinLength {
```

Comment:
It feels like we could save some code by making ExtremumLength be `ExtremumLength<T>` where `T` is either `None` or `Auto`. Not necessary though.